From: Chengguang Xu <cgxu519@gmx.com>
Date: Mon, 9 Jul 2018 22:48:07 +0800
Subject: ceph: return errors from posix_acl_equiv_mode() correctly
Git-commit: 61ad36d47dd273b7b8c3d63fc8359d96f7976c79
Patch-mainline: v4.19-rc1
References: bsc#1107320

In order to return correct error code should replace variable ret
using err in error case.

Signed-off-by: Chengguang Xu <cgxu519@gmx.com>
Reviewed-by: "Yan, Zheng" <zyan@redhat.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/acl.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/fs/ceph/acl.c
+++ b/fs/ceph/acl.c
@@ -168,10 +168,10 @@ int ceph_pre_init_acls(struct inode *dir
 		return err;
 
 	if (acl) {
-		int ret = posix_acl_equiv_mode(acl, mode);
-		if (ret < 0)
+		err = posix_acl_equiv_mode(acl, mode);
+		if (err < 0)
 			goto out_err;
-		if (ret == 0) {
+		if (err == 0) {
 			posix_acl_release(acl);
 			acl = NULL;
 		}
